### DISCUSSION TITLE
Junit test case for Acceleo

### DISCUSSION THREAD

**Discussion by Janmejaya1984** [eclipse-sirius/sirius-desktop#389] [General]
How to write and run junit test cases for acceleo template?

**Comment by ylussaud** [comment]
For Acceleo 3.x you have a [Acceleo Unit](https://github.com/eclipse-acceleo/acceleo/tree/R3_7_maintenance/unit/) update site. You can then use annotations to write your unit tests (an [example](https://github.com/eclipse-acceleo/acceleo/blob/R3_7_maintenance/unit/tests/org.eclipse.acceleo.unit.sample.tests/src/org/eclipse/acceleo/unit/sample/files/test/TemplatesGenerateFileTest.java)). There is a [post](https://www.tumblr.com/stephanebegaudeau/2379262223/having-fun-with-junit-and-acceleo) on how to use it.

For Acceleo 4 (for future readers) you can extends the following [test suite](https://github.com/eclipse-acceleo/acceleo/blob/master/tests/org.eclipse.acceleo.aql.tests/src/org/eclipse/acceleo/tests/utils/AbstractEvaluationTestSuite.java). For instance this [class](https://github.com/eclipse-acceleo/acceleo/blob/master/tests/org.eclipse.acceleo.aql.tests/src/org/eclipse/acceleo/tests/evaluation/FileStatementTests.java) and its [folder structure](https://github.com/eclipse-acceleo/acceleo/tree/master/tests/org.eclipse.acceleo.aql.tests/resources/evaluation/fileStatement). Each folder with the same file structure will be used as a test case.
